service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -248,11 +281,24 @@ protected ClientConfigProperties getDownstreamSenderFactoryConfigDefaults() {
      */
     @Qualifier(Constants.QUALIFIER_MESSAGING)
     @Bean
+    @ConditionalOnAmqpMessaging
     @Scope("prototype")
     public HonoConnection downstreamConnection() {
         return HonoConnection.newConnection(vertx(), downstreamSenderFactoryConfig());
     }
 
+    /**
+     * Exposes a factory for creating producers for sending downstream messages via the Kafka cluster.
+     *
+     * @return The factory.
+     */
+    @Bean
+    @Scope("prototype")
+    @ConditionalOnKafkaMessaging
+    public CachingKafkaProducerFactory<String, Buffer> kafkaProducerFactory() {
+        return CachingKafkaProducerFactory.sharedProducerFactory(vertx());
+    }
+
```

Comment:
can we keep the AMQP related beans together?

Reply:
Done.